client/my-sites/domains/domain-management/transfer/transfer-to-other-site/confirmation-dialog.jsx

```diff
@@ -49,24 +49,31 @@ class TransferConfirmationDialog extends PureComponent {
 	}
 
 	renderTargetSiteOnFreePlanWarnings() {
-		if ( this.props.primaryWithPlansOnly ) {
-			return (
+		const { isMapping } = this.props;
+
+		if ( isMapping ) {
+			return this.props.primaryWithPlansOnly ? (
 				<p>
 					{ this.props.translate(
 						"The target site doesn't have a paid plan, so you'll have to pay the full price for a " +
-							'domain mapping subscription when the domain mapping next renews. You will not be able to set it as primary either. ' +
+							'domain connection subscription when the domain connection next renews. You will not be able to set it as primary either. ' +
 							'If you upgrade the target site to a paid plan, these features are included in the plan.'
 					) }
 				</p>
+			) : (
+				<p>
+					{ this.props.translate(
+						"The target site doesn't have a paid plan, so you'll have to pay the full price for a " +
+							'domain connection subscription when the domain connection next renews. ' +
+							'If you upgrade the target site to a paid plan, domain connections (and many more features!) are included in our plans.'
+					) }
+				</p>
 			);
 		}
-
 		return (
 			<p>
 				{ this.props.translate(
-					"The target site doesn't have a paid plan, so you'll have to pay the full price for a " +
-						'domain mapping subscription when the domain mapping next renews. ' +
-						'If you upgrade the target site to a paid plan, domain mappings (and many more features!) are included in our plans.'
+					"The target site doesn't have a paid plan, so you won't be able to set this domain as primary on the site."
 				) }
 			</p>
 		);
```

Comment:
This should only be returned when `this.props.primaryWithPlansOnly` is `true`.

So I think we can do something like:
```
		if ( this.props.primaryWithPlansOnly ) {
			return (
				<p>
					{ this.props.translate(
						"The target site doesn't have a paid plan, so you won't be able to set this domain as primary on the site."
					) }
				</p>
			);
		}

		return null;
```